napari/_qt/widgets/qt_keyboard_settings.py

```diff
@@ -568,16 +563,14 @@ def keyPressEvent(self, event):
             self.setText('')
             return
 
-        key_map = {
-            Qt.Key.Key_Control: keys.CONTROL.name,
-            Qt.Key.Key_Shift: keys.SHIFT.name,
-            Qt.Key.Key_Alt: keys.ALT.name,
-            Qt.Key.Key_Meta: keys.META.name,
-            Qt.Key.Key_Delete: keys.DELETE.name,
-        }
-
-        if event_key in key_map:
-            self.setText(key_map[event_key])
+        if event_key in (
+            Qt.Key.Key_Control,
+            Qt.Key.Key_Shift,
+            Qt.Key.Key_Alt,
+            Qt.Key.Key_Meta,
+            Qt.Key.Key_Delete,
+        ):
```

Comment:
```suggestion
        if event_key in {
            Qt.Key.Key_Control,
            Qt.Key.Key_Shift,
            Qt.Key.Key_Alt,
            Qt.Key.Key_Meta,
            Qt.Key.Key_Delete,
        }:
```

Reply:
what's the advantage of switching to a set instead of a tuple here?

Reply:
It should be speedup and style. 

But I do some additional tests and it looks like in this case the speedup is not observed:
```
In [14]: %timeit Qt.Key.Key_Control in (Qt.Key.Key_Control, Qt.Key.Key_Shift, Qt.Key.Key_Alt, Qt.Key.Key_Meta, Qt.Key.Key_Delete,)
6.84 µs ± 380 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [15]: %timeit Qt.Key.Key_Delete in (Qt.Key.Key_Control, Qt.Key.Key_Shift, Qt.Key.Key_Alt, Qt.Key.Key_Meta, Qt.Key.Key_Delete,)
6.95 µs ± 131 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [16]: %timeit Qt.Key.Key_Control in {Qt.Key.Key_Control, Qt.Key.Key_Shift, Qt.Key.Key_Alt, Qt.Key.Key_Meta, Qt.Key.Key_Delete,}
6.83 µs ± 159 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [17]: %timeit Qt.Key.Key_Delete in {Qt.Key.Key_Control, Qt.Key.Key_Shift, Qt.Key.Key_Alt, Qt.Key.Key_Meta, Qt.Key.Key_Delete,}
7.05 µs ± 323 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

The additional test shows that creation time is essential:

```
In [18]: a = {Qt.Key.Key_Control, Qt.Key.Key_Shift, Qt.Key.Key_Alt, Qt.Key.Key_Meta, Qt.Key.Key_Delete,}

In [19]: b = (Qt.Key.Key_Control, Qt.Key.Key_Shift, Qt.Key.Key_Alt, Qt.Key.Key_Meta, Qt.Key.Key_Delete,)

In [20]: %timeit Qt.Key.Key_Delete in a
1.04 µs ± 29.5 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [21]: %timeit Qt.Key.Key_Delete in b
1.08 µs ± 22.5 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [22]: %timeit Qt.Key.Key_Control in b
966 ns ± 43.4 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [23]: %timeit Qt.Key.Key_Control in a
958 ns ± 18 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```

So next test:

```
In [10]: event_key = Qt.Key.Key_Control

In [24]: %timeit event_key in a
52.8 ns ± 4.48 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [25]: %timeit event_key in b
41 ns ± 0.744 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [26]: event_key = Qt.Key.Key_Delete

In [27]: %timeit event_key in a
47.4 ns ± 0.934 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [28]: %timeit event_key in b
76.6 ns ± 0.942 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```

but put this in function does not help:
```
In [34]: def check_set(key):
    ...:     return key in {Qt.Key.Key_Control, Qt.Key.Key_Shift, Qt.Key.Key_Alt, Qt.Key.Key_Meta, Qt.Key.Key_Delete,}
    ...: 

In [35]: def check_tuple(key):
    ...:     return key in (Qt.Key.Key_Control, Qt.Key.Key_Shift, Qt.Key.Key_Alt, Qt.Key.Key_Meta, Qt.Key.Key_Delete,)
    ...: 

In [36]: event_key = Qt.Key.Key_Delete

In [37]: %timeit check_set(event_key)
5.96 µs ± 157 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [38]: %timeit check_tuple(event_key)
5.84 µs ± 219 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [39]: event_key = Qt.Key.Key_Control

In [40]: %timeit check_set(event_key)
5.85 µs ± 104 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [41]: %timeit check_tuple(event_key)
5.81 µs ± 136 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

So finaly checks show that access to qt enum members is slow 

```
In [50]: %timeit Qt.Key.Key_Control == Qt.Key.Key_Control
1.86 µs ± 20.1 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [51]: %timeit hash(event_key)
54.8 ns ± 2.46 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [52]: %timeit hash(Qt.Key.Key_Control)
958 ns ± 15.8 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [53]: %timeit event_key == event_key
34.2 ns ± 0.577 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```

So everything that comes from python data structures is not important in access to qt enums. 

So only style left.

Reply:
ok, I can put this set in a constant then